### PR TITLE
Multiple fallback languages

### PR DESCRIPTION
**Pull request by Chive**
Hello.

* Work in progress (just grep'ed through the whole codebase and did replacements where neccessary)
* Tests are passing (except for the ones failing in your master anyway)
* Docs are not adapted (yet)

Would like to hear opinions.

Best

Chive

### CODE REVIEW

File: parler/models.py [hunk 1]
```diff
@@ -307,11 +307,20 @@ def set_current_language(self, language_code, initialize=False):
 
     def get_fallback_language(self):
         """
+        Backwards compatible fallback language getter
+        """
+        fallbacks = self.get_fallback_languages()
+        return fallbacks[0] if fallbacks else None
```
Comment (vdboor) [inline parler/models.py]: +1 for keeping compatibility.

File: parler/appsettings.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
 # Have to fill the default section explicitly to avoid circular imports
 PARLER_LANGUAGES.setdefault('default', {})
 PARLER_LANGUAGES['default'].setdefault('code', PARLER_DEFAULT_LANGUAGE_CODE)
-PARLER_LANGUAGES['default'].setdefault('fallback', PARLER_DEFAULT_LANGUAGE_CODE)
+PARLER_LANGUAGES['default'].setdefault('fallbacks', [PARLER_DEFAULT_LANGUAGE_CODE])
```
Comment (vdboor) [inline parler/appsettings.py]: I think this should take care of the fact that people may still have `'fallback': ..` in their config.

File: parler/models.py [hunk 3]
```diff
@@ -307,11 +307,20 @@ def set_current_language(self, language_code, initialize=False):
...
+
+    def get_fallback_languages(self):
+        """
         Return the fallback language code,
         which is used in case there is no translation for the currently active language.
         """
         lang_dict = get_language_settings(self._current_language)
-        return lang_dict['fallback'] if lang_dict['fallback'] != self._current_language else None
+        fallbacks = [lang for lang in lang_dict['fallbacks'] if lang != self._current_language]
+        return fallbacks or None
```
Comment (vdboor) [inline parler/models.py]: I think this should be `or []` so everything else can just loop through the list.

File: parler/models.py [hunk 4]
```diff
@@ -466,13 +475,14 @@ def _get_translated_model(self, language_code=None, use_fallback=False, auto_cre
...
             # Don't cache under this language_code
-            try:
-                return self._get_translated_model(lang_dict['fallback'], use_fallback=False, auto_create=auto_create, meta=meta)
-            except meta.model.DoesNotExist:
-                fallback_msg = " (tried fallback {0})".format(lang_dict['fallback'])
+            for fallback_lang in lang_dict['fallbacks']:
+                try:
+                    return self._get_translated_model(fallback_lang, use_fallback=False, auto_create=auto_create, meta=meta)
+                except meta.model.DoesNotExist:
+                    fallback_msg = " (tried fallbacks {0})".format(', '.join(lang_dict['fallbacks']))
```
Comment (vdboor) [inline parler/models.py]: I think this can be made a bit cleaner, and more precise. e.g. only set this message when all fallbacks failed.

**Comment by vdboor** [comment]
Thanks for this work, I'm really happy to see this being implemented in such clean way, and also taking care of backwards compatibility. I've added some notes for review, and the main idea looks pretty good!

File: parler/appsettings.py (hunk 2 above)
Comment (Chive) [inline parler/appsettings.py]: What would you like to see here? Should we raise a Deprecation Warning? Or just print something in the console?

File: parler/models.py (hunk 3 above)
Comment (Chive) [inline parler/models.py]: +1

**Comment by Chive** [comment]
Thanks! Adressed your points, not sure about ``parler/appsettings.py:20`` though.

File: parler/models.py (hunk 4 above)
Comment (Chive) [inline parler/models.py]: Totally agree, changed it

**Comment by vdboor** [comment]
Thanks!

**Comment by mkoistinen** [comment]
This really needs to be released! Its a life-saver!

**Comment by vdboor** [comment]
FYI, it's out! django-parler 1.5 was released last week, under the encouragement of a email from Divio :)

**Comment by Chive** [comment]
Awesome, thanks @vdboor!

**Pull request merged**